arrow-docs/docs/docs/datatypes/intro/README.md

```diff
@@ -0,0 +1,148 @@
+so there is a function `fold` which takes a continuation function per case: presence or absence.
+The implementation of `fold()` is a simple `when` to check whether `this` is a `Some` or a `None` and apply the appropriate continuation function.
+Whether `Option` works differently for `String` or `Int` or absence is up to the user.
+
+All other functions provided by option are implemented by using `fold()`, making for idiomatic helper functions like `getOrNull`, `getOrElse`, or `map`.
+Feel free to explore the implementation of `Option` and other datatypes to discover their behavior!
+
+### Datatypes in Arrow
+
+We will list all the datatypes available in arrow by the module they belong to, and a short description of the coding pattern they abstract.
+
+#### Core
+
+Core contains the datatypes that are also used by the public API of several [typeclasses]({{ '/docs/patterns/glossary/' | relative_url }}),
+so they are always required.
+
+- [`Id`]({{ '/docs/datatypes/id/' | relative_url }}) - a simple wrapper without any behavior, used mostly for testing
+
+- [`Option`]({{ '/docs/datatypes/option/' | relative_url }}) - absence of a value, or failure to construct a correct value
+
+- [`Either`]({{ '/docs/datatypes/either/' | relative_url }}) - an if/else branch in execution
+
+- [`Eval`]({{ '/docs/datatypes/eval/' | relative_url }}) - lazy evaluation of functions with stack safety and memoization
+
+- `TupleN` - a heterogeneous grouping of 2-9 values without creating a named class
+
+#### Data
+
+Data contains the bulk of the datatypes provided by Arrow. We can separate them onto several categories.
+
+##### General use
+
+- [`NonEmptyList`]({{ '/docs/datatypes/nonemptylist/' | relative_url }}) - a homogeneous list that has at least 1 value
+
+- [`Ior`]({{ '/docs/datatypes/ior/' | relative_url }}) - a branch in execution for three possible paths: one, two, or both
+
+- [`Const`]({{ '/docs/datatypes/const/' | relative_url }}) - tags a value with a "phantom generic" that's never instantiated, and it can be used for example to represents units or state
+
+- [`Coproduct`]({{ '/docs/datatypes/coproduct/' | relative_url }}) - constructs a new composed type from two datatypes, allowing to contain and operate on either one of them
+
+##### Error handling
+
+- [`Try`]({{ '/docs/datatypes/try/' | relative_url }}) - returns the result of executing a block of code that can fail and throw exceptions
+
+- [`Validated`]({{ '/docs/datatypes/validated/' | relative_url }}) - returns the result of aggregating multiple calculations that can fail, and it also aggregates the errors
+
+##### Reader/Writer/State
+
+- [`Kleisli`]({{ '/docs/datatypes/kleisli/' | relative_url }}) - similar to Dependency Injection and Inversion of Control, it represents a calculation with a dependency on an external context
+
+- [`Reader`]({{ '/docs/datatypes/reader/' | relative_url }}) - same as kleisli but tied to a single datatype
```

Comment:
`but operating over the Id data type`. Also I think the general accepted convention is `data type` and `type class` instead of `datatype/typeclass`

Reply:
Done

Reply:
Datatype/typeclass will go into another PR, I don't want to fix that right now haha